streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -590,7 +590,14 @@ boolean runLoop() {
                 log.warn("Detected the states of tasks " + e.corruptedTasks() + " are corrupted. " +
                          "Will close the task as dirty and re-create and bootstrap from scratch.", e);
                 try {
+                    // check if any active task got corrupted. We will trigger a rebalance in that case.
+                    // once the task corruptions have been handled
+                    Set<Task> corruptedActiveTasks = taskManager.anyActiveTasksCorrupted(e.corruptedTasks());
```

Comment:
nit: instead of trying to get the actual set of corrupted active tasks just to check whether its empty or not, how about we make `TaskManager#handleCorruption` return a boolean instead that tells the StreamThread whether or not to trigger a rebalance. For one thing we happen to already sort out the active tasks from the standbys during that method so we can figure it out from that rather than recomputing this in the thread. For another, it's probably a good idea to keep this logic consolidated in one place, in particular within the TaskManager, so we can easily expand on this condition eg to further optimize based on task metadata or so on. Thoughts?

Reply:
Yeah that makes sense. I had the same thought but wasn't sure if I should go ahead and change the return type. I have made the changes. The method name might need to change though- right now it's unchanged?